sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/impl/action/ActionEffectImpl.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.scalasdk.impl.action
+
+import scala.collection.immutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters.IterableHasAsJava
+
+import com.akkaserverless.javasdk
+import com.akkaserverless.scalasdk.Metadata
+import com.akkaserverless.scalasdk.ServiceCall
+import com.akkaserverless.scalasdk.SideEffect
+import com.akkaserverless.scalasdk.action.Action
+import com.akkaserverless.scalasdk.impl.JavaServiceCallAdapter
+import com.akkaserverless.scalasdk.impl.JavaSideEffectAdapter
+
+object ActionEffectImpl {
```

Comment:
I wonder if you can't just delegate to java impl, like in `scalasdk.impl.ValueEntityEffectImpl` (not a big deal, but might mean less duplication)

Reply:
We can explore this eventually on another PR. I just realised that Value Entities and ES Entities are encoded quite different than Actions. 

The `Impl` class is a Builder and an Effect. This is not how the Actions are implemented though. There are different `Effect` implementations.